python/ray/data/_internal/logical/rules/operator_fusion.py

```diff
@@ -479,6 +484,7 @@ def fused_on_start(schema):
                 fn_constructor_kwargs=down_logical_op._fn_constructor_kwargs,
                 min_rows_per_bundled_input=min_rows_per_bundled_input,
                 compute=compute,
+                can_modify_num_rows=can_modify_num_rows,
```

Comment:
I understand your intention here Justin, but the actual problem here is that the impl of the current rule isn't right -- instead of this hoopla with carrying logical op, we should split this rule into 2: one applied at logical level, another at physical.

Let's try to see if splitting it up is feasible immediately and if it's let's do that. If it's not, we can come back to this as temporary solution